aten/src/ATen/native/mps/operations/UnaryKernel.mm

```diff
@@ -94,13 +94,11 @@
   }
   using namespace mps;
   @autoreleasepool {
-    Tensor inputTensor = self;
     id<MTLDevice> device = MPSDevice::getInstance()->device();
     id<MTLComputePipelineState> cplState =
         getCPLState(device, getMetalType(outputTensor), getMetalType(self), "erfinv_mps_kernel");
 
     if (!self.is_contiguous()) {
-      inputTensor = inputTensor.contiguous();
```

Comment:
Curious why is it removed? I feel we are gonna need this for a non-contiguous input tensor. Or, alternatively, we should add a kernel_index_offset PSO that precedes the erfinv kernel to calculate the strides of the input tensor.

https://github.com/pytorch/pytorch/blob/04f20bb28581c2a3895c09feeebb37401b0c2fc7/aten/src/ATen/native/mps/operations/HistogramKernel.mm#L271-L286

Reply:
Interesting, yes, we should add a test case with view ops as well to test it.

Reply:
Thank you for the feedback. I'll work on this soon.

Reply:
> Curious why is it removed? I feel we are gonna need this for a non-contiguous input tensor. Or, alternatively, we should add a kernel_index_offset PSO that precedes the erfinv kernel to calculate the strides of the input tensor.
> 
> https://github.com/pytorch/pytorch/blob/04f20bb28581c2a3895c09feeebb37401b0c2fc7/aten/src/ATen/native/mps/operations/HistogramKernel.mm#L271-L286

I did some basic test case intially and it was working without it so I thought it was not needed. But further testing with more complicated shape per my new slice test in the updated test_mps.py would cause it to fail. Therefore, I added it back in as you suggested to fix the problem.